lib/workers/repository/update/pr/labels.ts

```diff
@@ -2,15 +2,29 @@ import is from '@sindresorhus/is';
 import { dequal } from 'dequal';
 import type { RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
+import { platform } from '../../../../modules/platform';
 import * as template from '../../../../util/template';
 
+/**
+ * Filter labels that go over the maximum char limit, based on platform limits.
+ */
+function trimLabel(label: string, limit: number): string {
+  if (label.length > limit) {
+    return label.trim().slice(0, limit).trim();
+  }
+
+  return label.trim();
```

Comment:
```suggestion
  const trimmed = label.trim();
  if (trimmed.length <= limit) {
    return trimmed;
  }

  return trimmed.slice(0, limit).trim();
```
This will correctly trim in case of dangling whitespace.